### PR TITLE
enh: add ability to hide sidebar in app layout

### PR DESCRIPTION
**Pull request by fontanierh**
for the "fullscreen modal" feel

use this new layout for:
- assistant builder
- apps (dev tools)

Other pages need to first be moved to the new layout
 

fixes https://github.com/orgs/dust-tt/projects/3?pane=issue&itemId=39794159

### CODE REVIEW

File: front/components/sparkle/AppLayout.tsx [hunk 1]
```diff
@@ -170,6 +171,7 @@ export default function AppLayout({
   navChildren?: React.ReactNode;
   titleChildren?: React.ReactNode;
   children: React.ReactNode;
+  hideSidebar?: boolean;
```
Comment (spolu) [inline front/components/sparkle/AppLayout.tsx:174]: let's mege this with isWideMode to `mode = "wide" | "full" | "normal"`

File: front/components/sparkle/AppLayout.tsx [hunk 2]
```diff
@@ -281,7 +290,9 @@ export default function AppLayout({
           </div>
           <div
             className={classNames(
-              "fixed left-0 right-0 top-0 z-30 flex h-16 flex-row pl-12 lg:left-80 lg:pl-0",
+              "fixed left-0 right-0 top-0 z-30 flex h-16 flex-row pl-12",
+              !hideSidebar ? "lg:left-80" : "",
+              "lg:pl-0",
```
Comment (spolu) [inline front/components/sparkle/AppLayout.tsx:295]: Why separate this one?

**Review by spolu — CHANGES_REQUESTED** [review]
LGTM but would love to collapse isWide and hideSideBar to one mode parameter?

File: front/components/sparkle/AppLayout.tsx (hunk 1 above)
Comment (fontanierh) [reply front/components/sparkle/AppLayout.tsx:174]: as per IRL, keeping them separate as you could want full + wide or full + not wide (and wide + not full)

File: front/components/sparkle/AppLayout.tsx (hunk 2 above)
Comment (fontanierh) [reply front/components/sparkle/AppLayout.tsx:295]: fixed

**Comment by fontanierh** [comment]
Moved the params close together + fixed the isolated classname @spolu

**Review by spolu — APPROVED** [review]

**Pull request merged**